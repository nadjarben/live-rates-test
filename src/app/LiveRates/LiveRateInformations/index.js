import React from "react";
import { convertFormatDate } from "../../../helpers";

const LiveRateInformations = ({
  liveRate: { 
    currency, 
    rate, 
    bid, 
    ask, 
    low, 
    high, 
    open, 
    close, 
    timestamp 
  },
}) => {
  const relativeDate = React.useCallback(() => {
    return convertFormatDate(timestamp)
  }, [timestamp])

  return (
    <tr>
      <td>{currency}</td>
      <td>{rate}</td>
      <td>{bid}</td>
      <td>{ask}</td>
      <td>{high}</td>
      <td>{low}</td>
      <td>{open}</td>
      <td>{close}</td>
      <td>{relativeDate()}</td>
    </tr>
  );
};

export default LiveRateInformations;
