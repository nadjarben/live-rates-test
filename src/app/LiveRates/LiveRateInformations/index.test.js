import React from "react";
import { Provider } from "react-redux";
import { store } from "../../../store";
import LiveRateInformations from "./";


const ShallowLiveRateInformations = (
  <Provider store={store}>
    <LiveRateInformations />
  </Provider>
);

describe("LiveRateInformations", () => {
  test("Does match snapshots", () => {
    expect(ShallowLiveRateInformations).toMatchSnapshot();
  });

  test("Does LiveRateInformations be defined", () => {
    expect(ShallowLiveRateInformations).toBeDefined();
  });
});
