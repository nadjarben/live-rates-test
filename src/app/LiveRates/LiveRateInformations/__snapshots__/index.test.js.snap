// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveRateInformations Does match snapshots 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <LiveRateInformations />
</Provider>
`;
